internal/storage/bucket_handle.go

```diff
@@ -280,6 +274,14 @@ func (b *bucketHandle) ListObjects(ctx context.Context, req *gcs.ListObjectsRequ
 			currObject := storageutil.ObjectAttrsToBucketObject(attrs)
 			list.Objects = append(list.Objects, currObject)
 		}
+
+		// itr.next returns all the objects present in the bucket. Hence adding a
+		// check to break after iterating over the current page.
+		// If req.MaxResults is 0, then wait till iterator is done. This is similar
+		// to https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/vendor/github.com/jacobsa/gcloud/gcs/bucket.go#L164
+		if req.MaxResults != 0 && (pi.Remaining() == 0) {
```

Comment:
can you also add a comment on what pi.remaining and why is it added at the end and not in the begining of the loop